website/blog/2024-05-17-whats-next.md

```diff
@@ -0,0 +1,68 @@
+
+Now after signing up to the poll with this new key, there will not be any link to the original key, and users will effectively be anonymous. Of course users should ensure that they are using different wallets where possible.
+
+Finally, with the use of a nullifier, it will not be possible for the same original key to be used to signup more than once for each new poll.
+
+Are there any drawbacks? Well, yes. There is now an extra step for users to register to individual Polls. We aim to offset this cost and additional step soon either with gasless transactions or by moving some logic offchain.
+
+## Message structure optimisation
+
+On top of the improvements to anonymity, the 3327 team aims to also replace the Merkle Tree used for storing messages with a hash chain. Some of the benefits of this approach are:
+
+- unlimited number of messages
+- removal of expensive merge operations from the coordinator
+- cheaper to send messages as only one hash is required to update the hash chain
+- less constraints on the circuits due to simplified logic
+
+**Unlimited messages**
+
+Merkle trees are usually bound by a depth property. Together with the number of leaves per node, we can calculate the max capacity of a tree. For instance, for a binary tree with a depth of 10, we can host up to **2^10** (1024) leaves. On the other hand, hash chains do not have a limit, unless if we wanted to set one, so we technically can support an unlimited number of messages.
+
+**Cheaper operations**
+
+In general, hashing the previous hash chain with the message is cheaper than inserting into a merkle tree, as well as the coordinator not needing to perform any merge operations on the accumulator queues that were used on chain, thus greatly reducing costs and processing time.
+
+**Smaller circuits**
+
+As cited in their proposal, processing message inclusion proofs for **k** messages in a tree with height **h** requires **k \* h** hashing operations within the circuit with **2 \* k \* h** signal values for inclusion proofs. Processing messages with chain hashes removes the unnecessary inclusion proofs and requires only **k** hashes to be computed for **k** messages without any extra signals, as the requirement is to prove that the order and inclusion of all messages are correct.
+
+## What does this mean for you, Anon?
```

Comment:
```suggestion
## A call for MACI grant proposals
```
So what does this mean for you, Anon?

As an open-source project of [PSE](https://pse.dev/) with support from the [Ethereum Foundation](https://ethereum.foundation/), MACI is fortunate to have the resources to invest in the maintenance and improvement of the protocol. This means we're able to fund full-time developers as well as allocate grants for various research and development initiatives.